index.js

```diff
@@ -123,17 +171,30 @@ module.exports = options => {
 
   /** Takes a new command and returns the result of evaluating it in the current context. */
   return rawCommand => {
-    const command = rawCommand.trim()
+    const commandTrimmed = rawCommand.trim()
+    const command = commandTrimmed.endsWith(';') ? 
+      commandTrimmed.substring(0, commandTrimmed.length - 1) : 
+      commandTrimmed
 
     // ignore blank lines
     if (command === '') {
       return Promise.resolve(null)
     }
 
-    const commandWithSemi = command + (command.endsWith(';') ? '' : ';')
-    return evalSol(commands.concat(commandWithSemi), options)
+    return evalSol(commands.concat(command + ";"), options)
       .then(result => {
-        commands.push(commandWithSemi)
+        commands.push(command + ";")
+        if (command in specialGlobals) {
+          let struct = {}
+
+          // result is an object, so we first convert it to string and then split
+          let results = result.toString().split(",")
```

Comment:
Is the `.toString` actually needed here?

Reply:
It is not enough to know that it breaks without `toString`, we need to know *why* it breaks. `toString` is not a solution; it is rarely correct to use `toString` in Javascript. We need to understand what type `result` is and act on the object appropriately, deconstructing it if necessary. Use `typeof` and `instanceof` during debugging to attempt to determine what `result` actually is and how we can make an array of it. I would almost think `result` is already an array if we have to split the `toString` output on commas again.